mailpoet/lib/Newsletter/Sending/ScheduledTaskSubscribersRepository.php

```diff
@@ -53,4 +54,45 @@ public function createOrUpdate(array $data): ?ScheduledTaskSubscriberEntity {
     $this->flush();
     return $taskSubscriber;
   }
+
+  public function countSubscriberIdsBatchForTask(int $taskId, int $lastProcessedSubscriberId): int {
+    $queryBuilder = $this->entityManager
+      ->createQueryBuilder()
+      ->select('count(sts.subscriber)');
+    $queryBuilder = $this->prepareSubscriberIdsBatchForTaskQuery($queryBuilder, $taskId, $lastProcessedSubscriberId);
+    $countSubscribers = $queryBuilder
+      ->getQuery()
+      ->getSingleScalarResult();
+
+    if (is_numeric($countSubscribers)) {
+      return (int)$countSubscribers;
+    } else {
+      return 0;
+    }
+  }
+
+  public function getSubscriberIdsBatchForTask(int $taskId, int $lastProcessedSubscriberId, int $limit): array {
+    $queryBuilder = $this->entityManager
+      ->createQueryBuilder()
+      ->select('IDENTITY(sts.subscriber) AS subscriber_id');
+    $queryBuilder = $this->prepareSubscriberIdsBatchForTaskQuery($queryBuilder, $taskId, $lastProcessedSubscriberId);
+    $subscribersIds = $queryBuilder
+      ->orderBy('sts.subscriber', 'asc')
+      ->setMaxResults($limit)
+      ->getQuery()
+      ->getSingleColumnResult();
+
+    return $subscribersIds;
+  }
+
+  private function prepareSubscriberIdsBatchForTaskQuery(QueryBuilder $queryBuilder, int $taskId, int $lastProcessedSubscriberId): QueryBuilder {
```

Comment:
This method could also contain creating a query builder because you avoid one-line code repetition, and the selection can be configured later, but it's a detail only.

Reply:
Good point, I was working under the incorrect assumption that I had to call `select()` before `from()`. I added a new commit implementing your suggestion. Thanks.